frontend/src/pages/artist/_itemId/index.vue

```diff
@@ -123,136 +119,120 @@ import {
+
+const artistBackdrop = computed(() =>
+  getImageInfo(item.value, { preferBackdrop: true })
+);
 
-    const discography =
-      (
-        await remote.sdk.newUserApi(getItemsApi).getItems({
-          albumArtistIds: [itemId],
-          sortBy: ['PremiereDate', 'ProductionYear', 'SortName'],
-          sortOrder: [SortOrder.Descending],
-          recursive: true,
-          includeItemTypes: [BaseItemKind.MusicAlbum],
-          fields: Object.values(ItemFields),
-          userId: remote.auth.currentUserId
-        })
-      ).data.Items ?? [];
+const singles = computed<BaseItemDto[]>(() =>
+  discography.value.filter(
+    (album) =>
+      (album?.RunTimeTicks ?? album?.CumulativeRunTimeTicks ?? 0) <=
+      msToTicks(SINGLE_MAX_LENGTH_MS)
+  )
+);
 
-    const singles = discography.filter(
-      (album: BaseItemDto) =>
-        (album?.RunTimeTicks || album?.CumulativeRunTimeTicks || 0) <=
-        msToTicks(albumBreakpoints.singleMsMaxLength)
-    );
-    const eps = discography.filter(
-      (album: BaseItemDto) =>
-        (album?.RunTimeTicks || album?.CumulativeRunTimeTicks || 0) >
-          msToTicks(albumBreakpoints.singleMsMaxLength) &&
-        (album?.RunTimeTicks || album?.CumulativeRunTimeTicks || 0) <=
-          msToTicks(albumBreakpoints.epMsMaxLength)
-    );
-    const albums = discography.filter(
-      (album: BaseItemDto) =>
-        (album?.RunTimeTicks || album?.CumulativeRunTimeTicks || 0) >
-        msToTicks(albumBreakpoints.epMsMaxLength)
-    );
+const eps = computed(() =>
+  discography.value.filter(
+    (album) =>
+      (album?.RunTimeTicks ?? album?.CumulativeRunTimeTicks ?? 0) >
+        msToTicks(SINGLE_MAX_LENGTH_MS) &&
+      (album?.RunTimeTicks ?? album?.CumulativeRunTimeTicks ?? 0) <=
+        msToTicks(EP_MAX_LENGTH_MS)
+  )
+);
 
-    const appearances =
-      (
-        await remote.sdk.newUserApi(getItemsApi).getItems({
-          contributingArtistIds: [itemId],
-          excludeItemIds: [itemId],
-          sortBy: ['PremiereDate', 'ProductionYear', 'SortName'],
-          sortOrder: [SortOrder.Descending],
-          recursive: true,
-          includeItemTypes: [BaseItemKind.MusicAlbum],
-          fields: Object.values(ItemFields),
-          userId: remote.auth.currentUserId
-        })
-      ).data.Items ?? [];
+const albums = computed(() =>
+  discography.value.filter(
+    (album) =>
+      (album?.RunTimeTicks ?? album?.CumulativeRunTimeTicks ?? 0) >
+      msToTicks(EP_MAX_LENGTH_MS)
+  )
+);
 
-    const musicVideo =
-      (
-        await remote.sdk.newUserApi(getItemsApi).getItems({
-          artistIds: [itemId],
-          sortBy: ['PremiereDate', 'ProductionYear', 'SortName'],
-          sortOrder: [SortOrder.Descending],
-          recursive: true,
-          includeItemTypes: [BaseItemKind.MusicVideo],
-          fields: Object.values(ItemFields),
-          userId: remote.auth.currentUserId
-        })
-      ).data.Items ?? [];
+onMounted(async () => {
```

Comment:
Any reason why not an async function call at the top level?

Reply:
I'm not sure that will work since the object return when `getItemById` is only reactive if it's found, and a straight undefined value if not. Going to leave this outside of the scope for this PR.